root/static/scripts/release-editor/recordingAssociation.js

```diff
@@ -306,7 +305,9 @@ function setSuggestedRecordings(track, recordings) {
     var lastRecording = track.recording.saved;
 
     if (!track.hasExistingRecording() && lastRecording) {
-        recordings = union([lastRecording], recordings);
+        recordings = [lastRecording].concat(
+            recordings.filter(x => x !== lastRecording),
+        );
```

Comment:
Good question. It might be slightly faster but I don't think performance matters here. It does have to stay an array so maybe that was why, but `[...new Set([lastRecording, ...recordings])]` should be fine.